src/proof/arith_proof.cpp

```diff
@@ -91,9 +100,10 @@ Node ProofArith::toStreamRecLFSC(std::ostream& out, TheoryProof * tp, theory::eq
     Assert(pf->d_children.size() >= 2);
 
     int neg = -1;
-    theory::eq::EqProof subTrans;
-    subTrans.d_id = theory::eq::MERGED_THROUGH_TRANS;
-    subTrans.d_node = pf->d_node;
+    std::shared_ptr<theory::eq::EqProof> subTrans =
+        std::make_shared<theory::eq::EqProof>();
```

Comment:
This pattern of creating an `EqProof` and then immediately setting its fields is repeated multiple times throughout the code. It seems like it would be cleaner to just add a constructor that expects an id and a node.

Reply:
This is a good idea, but it is outside of the scope of this PR.